internal/nginx/config/generator.go

```diff
@@ -150,6 +174,60 @@ func generateProxyPass(address string) string {
 	return "http://" + address
 }
 
+var supportedFilters = map[v1beta1.HTTPRouteFilterType]struct{}{
+	v1beta1.HTTPRouteFilterRequestRedirect: {},
+}
+
+func findFirstFilters(
```

Comment:
Why handle this in the generator?

Would it make sense to move this processing to the state package and create an internal representation of a filter? I think that would make it easier to report status on invalid filters in the future.

Reply:
> I think that would make it easier to report status on invalid filters in the future.

that makes sense. let me experiment. I'll share a prototype shortly.

Reply:
here is a prototype for adding validation -- https://github.com/nginxinc/nginx-kubernetes-gateway/commit/918584e809c7e04ffdd2538e862383aa02257e7d

here is making Redirect filter part of the MatchRule of the state package -- https://github.com/nginxinc/nginx-kubernetes-gateway/commit/181073193b0ae9e7528ac3f3e9f969ed00fc3f3f -- which is done on top of the validation prototype

let me know what you think

Reply:
made filters part of configuration https://github.com/nginxinc/nginx-kubernetes-gateway/pull/218/commits/875b213265cfdc12f0a16b0b3bdd04dace9417a2